src/components/MenuButton/index.tsx

```diff
@@ -1,147 +1,51 @@
-import styled from '@emotion/styled'
-import {
-  CustomDropdown,
-  CustomDropdownItem as Item,
-  CustomDropdownPostion,
-} from '../CustomDropdown'
-import { color } from '../../theme'
 import {
-  useKeyPress,
-  useOnClickOutside,
-  usePrevious,
-  useTransition,
-} from '../../hooks'
-import { TransitionState } from '../../hooks/useTransition'
-
-interface MenuButtonItemPropType {
-  name: string
-  value: string
-}
-
-export interface MenuButtonProps {
-  dropdownPosition?: CustomDropdownPostion
-  items: MenuButtonItemPropType[]
-  onChange: (selectedItem: MenuButtonItemPropType) => void
-  initialSelectedItem: MenuButtonItemPropType
-}
-
-const duration = 200
-
-const Wrapper = styled.div`
-  display: inline-block;
-  position: relative;
-`
+  Menu as ReachMenu,
+  MenuList as ReachMenuList,
+  MenuItem as ReachMenuItem,
+  MenuLink as ReachMenuLink,
+} from '@reach/menu-button'
+import styled from '@emotion/styled'
+import { color, radius, shadow, space } from '../../theme'
+import React from 'react'
+import { css, Global } from '@emotion/react'
 
-interface StyledButton {
-  clickThrough?: boolean
+export const Menu: React.FC = ({ children, ...props }) => {
+  return (
+    <ReachMenu {...props}>
+      {children}
+      <Global
+        styles={css`
+          :root {
+            --reach-menu-button: 1;
+          }
+        `}
+      />
+    </ReachMenu>
+  )
 }
 
-const Button = styled.button<StyledButton>`
-  font-size: inherit;
-  color: ${color.spaceLight};
-  outline: 0;
-  pointer-events: ${props => (props.clickThrough ? 'none' : 'false')};
+// @ts-ignore
+export const MenuList = styled(ReachMenuList)`
+  background-color: ${color.nova};
+  border-radius: ${radius.lg};
+  box-shadow: ${shadow.strong};
+  padding: ${space[16]} 0;
 
-  &:focus {
-    box-shadow: none;
+  [data-reach-menu-item][data-selected] {
+    background-color: ${color.stardust};
   }
 `
 
-interface StyledMenuContainer {
-  transitionState:
-    | TransitionState.ENTERED
-    | TransitionState.ENTERING
-    | TransitionState.EXITED
-    | TransitionState.EXITING
-    | boolean // @TODO: Why is `state` boolean | TransitionState?
-  dropdownPosition?: CustomDropdownPostion
-}
-
-const MenuContainer = styled.div<StyledMenuContainer>`
-  position: absolute;
-  z-index: 2;
-  left: 50%;
-  bottom: 0;
-  opacity: ${props =>
-    props.transitionState === 'entering' || props.transitionState === 'entered'
-      ? 1
-      : 0};
-  transform: ${props =>
-    props.transitionState === 'entering' ||
-    props.transitionState === 'entered' ||
-    props.transitionState === 'exiting'
-      ? 'translate3d(-50%,0,0)'
-      : `translate3d(-50%,${props.dropdownPosition === 'top' ? -1 : 1}rem,0)`};
-  transition: opacity ${duration}ms ease-out, transform ${duration}ms ease-out;
+// @ts-ignore
+export const MenuItem = styled(ReachMenuItem)`
+  padding: ${space[8]} ${space[16]};
+  cursor: pointer;
 `
 
-const Menu = styled.div`
-  display: inline-block;
+// @ts-ignore
```

Comment:
Why do we need the @ts-ignore?

Reply:
Very strange type collision between reach and emotion 🤔 
I haven't figured it out yet, so I will make an issue for it